### PR TITLE
Tweak Wh3's VCGen

### PR DESCRIPTION
**Pull request by jhjourdan**
This PR tweaks Why3 VCGen with two attributes:
- vc:sp uses the "sp" VC generator, which avoids generating as many VCs as there are paths in the program, and rather uses disjunctions. This makes VCs no longer exponential in the size of the program.
- vc:do_not_keep_trace disables "tracing", i.e., adding variables in the context that corresponds to values of variables of the program. Because MIR contains a lot of temporaries, tracing is very expensive for our formulas.

These two changes greatly improves the performance of loading the Why3 IDE.

### CODE REVIEW

**Comment by jhjourdan** [comment]
Sadly, `vc:do_not_keep_trace` as an attribute is ignored by Why3 for recursive declarations (see [here](https://gitlab.inria.fr/why3/why3/-/blob/master/src/mlw/vc.ml#L1705). We need to fix Why3.

**Comment by jhjourdan** [comment]
@xldenis, what do you think? One of my concerns is the impact this will have on CreuSAT.

**Comment by xldenis** [comment]
> @xldenis, what do you think? One of my concerns is the impact this will have on CreuSAT.

Hard for me to ancipate (cc @sarsko). I'm going to run the proofs now.

File: why3/src/mlcfg/printer.rs [hunk 1]
```diff
@@ -527,7 +527,9 @@ impl Print for CfgFunction {
             .append(if self.rec { "rec " } else { "" })
             .append("cfg ")
             .append(if self.constant { "constant " } else { "" })
-            .append(self.sig.pretty(alloc, env).append(alloc.line_()).append(alloc.text(" =")))
+            .append(
+                self.sig.pretty(alloc, env).append(alloc.line_()).append(alloc.text(" = [@vc:sp]")),
```
[the remark this answers is not in the document]
Comment (jhjourdan) [reply why3/src/mlcfg/printer.rs:531]: I'm not sure this is relevant here. This is not an expression.

[the remark this answers is not in the document]
Comment (xldenis) [reply why3/src/mlcfg/printer.rs:531]: Hmm, good point this is `mlcfg` we are talking about, it could still be part of the AST of `CfgFunction` though.

**Comment by xldenis** [comment]
I tried rebuilding CreuSAT using this branch without much luck. about a dozen functions fail to prove.  I attempted to manually push through some of the goals without much luck.

I think for now, if we want to introduce this it should be conditional.

**Comment by xldenis** [comment]
**Update:** I have rebuilt CreuSAT using this branch and all the proofs work. It doesn't _fully_ pass using "Auto Level 3", several proofs need a further call to level 3 or an `inline_all` to then succeed, but that was also the case before I can't tell if that occured more or less than before.

**Comment by jhjourdan** [comment]
Alright, so do we agree that this is a good change?

**Comment by xldenis** [comment]
I think so.

**Pull request merged**